WooCommerce/WooCommerceUITests/Tests/ProductsTests.swift

```diff
@@ -0,0 +1,29 @@
+import UITestsFoundation
+import XCTest
+
+final class ProductsTests: XCTestCase {
+
+    override func setUpWithError() throws {
+        continueAfterFailure = false
+
+        let app = XCUIApplication()
+        app.launchArguments = ["logout-at-launch", "disable-animations", "mocked-wpcom-api", "-ui_testing"]
+        app.launch()
+        try LoginFlow.logInWithWPcom()
+    }
+
+    func testProductsScreenLoad() throws {
+        try TabNavComponent().gotoProductsScreen()
+        XCTAssert(try ProductsScreen().isLoaded)
+
+        let products = try GetMocks.readProductsData()
+        let firstProductName = products.response.jsonBody.data[0].name
```

Comment:
I would see a great value in checking that all products we have from mocks are present, not only the first one.

Reply:
> What do you think about meeting in the middle:
>
> 1. Check the values for the first product
> 2. Check that the number of products displayed matches the one from the mocked data (response returned 5 > products, 5 products displayed on screen)

I think this is a good compromise as long as in `2` the test checks the actual values presence, not just the total cells
number (e.g. the case when same product is shown 5 times will be caught by the test). 

On a general note, I agree that execution time is a very important (and limiting) factor, and we should aim for making them as fast as possible. But, I'm not sure if this should be achieved by making the tests do less, and loose a part of confidence. I think the best way to speed up the tests is by improving their performance, which will allow us to do more in the same timeframe.

(btw, out of curiosity, I checked the time increase for the case of all products opening, and it was 20 seconds).

Reply:
Yup, the validation will verify that only 1 instance of the product name will appear on the list so that's covered.

I agree that we shouldn't be testing less to the point where confidence is lost, but I also believe that there should be a balance between what's tested and the cost/value of it (be it execution time or effort).